### PR TITLE
dry-run: Promote from Alpha to Beta

### PR DESCRIPTION
**Pull request by apelisse**
Also remove the test that used to check that the dry-run flag is
forbidden.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
Dry-run is promoted to Beta and will be enabled by default.
```

### CODE REVIEW

**Comment by lavalamp** [comment]
Is this the only change needed?

**Comment by apelisse** [comment]
We need to update the documentation, but the release branch hasn't been created yet. We don't want to change the default for the webhook configuration before GA, so I do think it is.

**Comment by apelisse** [comment]
See #68514

**Comment by lavalamp** [comment]
/approve
/lgtm

To give @deads2k / @smarterclayton / @liggitt a chance to weigh in:
/hold

File: staging/src/k8s.io/apiserver/pkg/endpoints/apiserver_test.go [hunk 1]
```diff
@@ -3865,55 +3865,6 @@ func (storage *SimpleRESTStorageWithDeleteCollection) DeleteCollection(ctx conte
 	return nil, nil
 }
 
-func TestDryRun(t *testing.T) {
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/endpoints/apiserver_test.go]: could keep this to make sure things still respect the feature gate being disabled until it is removed (with a call to `defer utilfeaturetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.DryRun, false)()`)

**Comment by liggitt** [comment]
other than the comment on the test, lgtm

File: staging/src/k8s.io/apiserver/pkg/endpoints/apiserver_test.go (hunk 1 above)
Comment (apelisse) [reply staging/src/k8s.io/apiserver/pkg/endpoints/apiserver_test.go]: Yes, agreed, makes sense to me

**Pull request merged**